### ISSUE TITLE
Manifest checksum error after first backup 

### ISSUE THREAD

**Issue by jalexandre0** [pgbackrest/pgbackrest#1201]
Please provide the following information when submitting an issue (feature requests or general comments can skip this):

1. pgBackRest version: 
    2.30

2. PostgreSQL version: 
    12.4

3. Operating system/version - if you have more than one server (for example, a database server, a repository host server, one or more standbys), please specify each: 
   7.8.2003, primary and standby

4. Did you install pgBackRest from source or from a package? 
    PGDG repository

5. Please attach the following as applicable:
    - `pgbackrest.conf` file(s)

 ```
[global]
process-max=8
repo-cipher-pass=*****************
repo-cipher-type=aes-256-cbc
repo-path=/
repo-s3-bucket=*****************
repo-s3-endpoint=s3-sa-east-1.amazonaws.com
repo-s3-key=*********************
repo-s3-key-secret=***********************************************
repo-s3-region=sa-east-1
repo-type=s3
retention-full=3
start-fast=y
stop-auto=y
backup-standby=y
repo1-s3-ca-file=/etc/pki/tls/certs/ca-bundle.crt
log-level-console=info
log-level-file=debug
archive-async=y
archive-push-queue-max=10GB
archive-get-queue-max=1GB
spool-path=/var/spool/pgbackrest


[global:archive-push]
compress-level=3

[dev-database]
db-path=/data/12/pgdata/

db2-path=/data/12/pgdata/
db2-host=primary.domain.com.br
```

    - `postgresql.conf` settings applicable to pgBackRest (`archive_command`, `archive_mode`, `listen_addresses`, `max_wal_senders`, `wal_level`, `port`)

```
       name       |                                          setting                                           
------------------+--------------------------------------------------------------------------------------------
 archive_command  | pgbackrest --log-level-console=info --process-max=8 --stanza=dev-database  archive-push %p
 archive_mode     | on
 listen_addresses | *
 max_wal_senders  | 40
 port             | 5432
 wal_level        | logical
(6 rows)
```

    - errors in the postgresql log file before or during the time you experienced the issue
No relevant errors found.
    
- log file in `/var/log/pgbackrest` for the commands run (e.g. `/var/log/pgbackrest/mystanza_backup.log`)

```
2020-10-19 10:40:04.701 P00  ERROR: [026]: unable to load backup manifest file '/backup/dev-database/20201016-073604F/backup.manifest' or '/backup/dev-database/20201016-073604F/backup.manifest.copy':
                                    ChecksumError: invalid checksum, actual 'adc2a1ab5ae136ccfa5e8e9933ad83dc82e710b5' but expected 'bbce4dc0ec63cbd87f9e4767e3cdfa76670022d9'
                                    ChecksumError: invalid checksum, actual 'adc2a1ab5ae136ccfa5e8e9933ad83dc82e710b5' but expected 'bbce4dc0ec63cbd87f9e4767e3cdfa76670022d9'
                                    --------------------------------------------------------------------
                                    If SUBMITTING AN ISSUE please provide the following information:
                                    
                                    version: 2.30
                                    command: restore
                                    options: --log-level-console=info --log-level-file=debug --pg2-host=dev-database5.dc.ifood.com.br --pg1-path=/data/restore-backrest --pg2-path=/data/12/pgdata --process-max=8 --repo1-cipher-pass=<redacted> --repo1-cipher-type=aes-256-cbc --repo1-path=/ --repo1-s3-bucket=devbackuppocnv --repo1-s3-ca-file=/etc/pki/tls/certs/ca-bundle.crt --repo1-s3-endpoint=s3.us-east-1.amazonaws.com --repo1-s3-key=<redacted> --repo1-s3-key-secret=<redacted> --repo1-s3-region=us-east-1 --repo1-type=s3 --stanza=dev-database
                                    
                                    stack trace:
                                    info/info:infoLoad:560:(trace log level required for parameters)
                                    info/manifest:manifestLoadFile:(storage: {type: s3, path: {"/"}, write: false}, fileName: {"<REPO:BACKUP>/20201016-073604F/backup.manifest"}, cipherType: 1)
                                    command/restore/restore:cmdRestore:(void)
                                    main:main:(debug log level required for parameters)
                                    --------------------------------------------------------------------
2020-10-19 10:40:04.701 P00 DETAIL: statistics: {"http.client":{"total":1},"http.request":{"total":3},"http.session":{"total":1},"socket.client":{"total":1},"socket.session":{"total":1},"tls.client":{"total":1},"tls.session":{"total":1}}
2020-10-19 10:40:04.701 P00   INFO: restore command end: aborted with exception [026]
2020-10-19 10:40:04.701 P00  DEBUG:     common/lock::lockRelease: (failOnNoLock: false)
2020-10-19 10:40:04.701 P00  DEBUG:     common/lock::lockRelease: => false
2020-10-19 10:40:04.701 P00  DEBUG:     common/exit::exitSafe: => 26
2020-10-19 10:40:04.701 P00  DEBUG:     main::main: => 26
```

7. Describe the issue:

After create a stanza and perform first backup, further operations dies with manifest checksum error. First time with this issue, occurs only when repo-cipher is active.

**Comment by dwsteele** [comment]
Did this work on the version you were running prior to 2.30? And it works with 2.30 as long as there is no repo encryption?

There must be something going on with how the manifest is being serialized/deserialized but I can't image what it is. Getting a copy of a manifest would be helpful. You can get it with:
```
pgbackrest repo-get backup/mail/20201016-073604F/backup.manifest
```
Feel free to redact the `cipher-pass` value contained in this file, but try not to change anything else.

Also, are there any special character encodings set on your server?

**Comment by dwsteele** [comment]
Corrected the command in the comment above to `pgbackrest repo-get backup/dev-database/20201016-073604F/backup.manifest`

**Comment by jalexandre0** [comment]
> Did this work on the version you were running prior to 2.30? And it works with 2.30 as long as there is no repo encryption?

Tried with 2.29 too. Clean backups works fine,  ciphered backups presents the  same error. 

> There must be something going on with how the manifest is being serialized/deserialized but I can't image what it is. Getting a copy of a manifest would be helpful. You can get it with:
> 
> ```
> pgbackrest repo-get backup/dev-database/20201016-073604F/backup.manifest
> ```
> 
> Feel free to redact the `cipher-pass` value contained in this file, but try not to change anything else.

Sorry, I need to redact some infos to make this public. 
I can provide the unaltered file via  secure channel, if it helps. Just provide the instructions. 

[backup.manifest.txt](https://github.com/pgbackrest/pgbackrest/files/5411199/backup.manifest.txt)


> Also, are there any special character encodings set on your server?

Operating System And database in en_US.UTF-8 

Thank you for the support.

**Comment by dwsteele** [comment]
> Sorry, I need to redact some infos to make this public.

No worries, what you gave me was enough to figure out what was wrong.

There are some database names (e.g. ` redacted-shop-catalog`) that start with spaces and our manifest format won't work with leading or trailing spaces in database names. This is clearly a bug, but I'm not sure how to fix it without breaking the manifest format.

I don't think this has anything to do with encryption so my guess is you have never backed up *this* cluster with pgBackRest. I don't see how it would have worked in any version, encrypted or not.

The immediate solution, if possible, is to remove the leading spaces from these database names. We'll also work on a fix, but that won't happen immediately.

**Comment by jalexandre0** [comment]
> The immediate solution, if possible, is to remove the leading spaces from these database names. We'll also work on a fix, but that won't happen immediately.

No problem. leading spaces in database name is a very odd use case. I certainly don't expect a quick or easy fix for that. 

I will discuss this issue with customer and see if they can rename those db's. 

Regards, Jefferson

**Comment by dwsteele** [comment]
> leading spaces in database name is a very odd use case.

It should really work, though. I did figure out how to fix this and maintain backward compatibility, so you can expect to see the fix in the next release, which is tentatively scheduled for Nov 9.

If these databases are not under your control it might be easier to wait.

**Comment by dwsteele** [comment]
This has been fixed in PR #1210. Feel free to have a look.
